src/pipeline.ts

```diff
@@ -245,17 +244,13 @@ export async function checkNewCandidateMatch(
 		hashesToExclude,
 	);
 
-	if (
-		assessment.decision !== Decision.MATCH &&
-		assessment.decision !== Decision.MATCH_SIZE_ONLY &&
-		assessment.decision !== Decision.MATCH_PARTIAL
-	) {
+	if (!DECISION_ALL_MATCH.includes(assessment.decision)) {
 		return null;
 	}
 
 	const result = await performAction(
 		assessment.metafile!,
-		assessment.decision,
+		assessment.decision as DecisionAllMatch,
```

Comment:
after making the `isDecisionAllMatch` change, this `as` shouldn't be necessary any more.